plugins/redirect.ts

```diff
@@ -1,12 +1,5 @@
 export default defineNuxtPlugin(() => {
   addRouteMiddleware((to) => {
-    if (to.path === '/products/detail') {
-      return navigateTo(
-        { name: 'products-details' },
-        { redirectCode: 301, replace: true },
-      );
-    }
-
```

Comment:
Instead of removing this can you make sure that both `/products/detail` and `/products/details` do a permanent redirect to `/products` instead?

It's better for search engines instead of going to `404`